### PR TITLE
[tplinksmarthome] Added KL430 lightstrip support

### LINKED ISSUE #8709: [tplinksmarthome] Add support for TP-Link KL430 Strip Light
There doesn't appear to be support for the Kasa KL430 strip light. I captured the following JSON between the Kasa app on an Android phone and a KL430 strip light with a KL430E extension. I removed what I think was the phone's id and replaced it with `<id>` below. The format of `<id>` was `<32-bits>-<16-bits>-<16-bits>-<16-bits>-<48-bits>`. I also replaced the real ssid with `<ssid>` below.

### Request
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":1,"transition":0}},"context":{"source":"<id>"}}`
### Response 
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":1,"mode":"normal","groups":[[0,31,0,0,100,4970]],"err_code":0}}}`

### Request
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":0,"transition":0}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":0,"dft_on_state":{"mode":"normal","groups":[[0,31,0,0,100,4970]]},"err_code":0}}}`

### Request
`{"smartlife.iot.lightStrip":{"set_light_state":{"brightness":73,"on_off":1,"transition":150}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":1,"mode":"normal","groups":[[0,31,0,0,73,4970]],"err_code":0}}}`

### Request
`{"smartlife.iot.lightStrip":{"set_light_state":{"color_temp":5275,"hue":0,"on_off":1,"saturation":0,"transition":150}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":1,"mode":"normal","groups":[[0,31,0,0,73,5275]],"err_code":0}}}`

### Request
`{"smartlife.iot.lightStrip":{"set_light_state":{"color_temp":0,"hue":115,"on_off":1,"saturation":75,"transition":150}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.lightStrip":{"set_light_state":{"on_off":1,"mode":"normal","groups":[[0,31,115,75,73,0]],"err_code":0}}}`

### Request
`{"smartlife.iot.lightStrip":{"apply_preset":{"index":1}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.lightStrip":{"apply_preset":{"err_code":0}}}`

### Request
`{"smartlife.iot.common.emeter":{"get_daystat":{"month":9,"year":2020}},"context":{"source":"<id>"}}`
### Response
`{"smartlife.iot.common.emeter":{"get_daystat":{"day_list":[{"year":2020,"month":9,"day":14,"energy_wh":141},{"year":2020,"month":9,"day":15,"energy_wh":257},{"year":2020,"month":9,"day":16,"energy_wh":264},{"year":2020,"month":9,"day":17,"energy_wh":283},{"year":2020,"month":9,"day":18,"energy_wh":240},{"year":2020,"month":9,"day":30,"energy_wh":1}],"err_code":0}}}`

### Request 1
`{"smartlife.iot.common.softaponboarding":{"get_stainfo":{}},"context":{"source":"<id>"}}`
### Response 1
`{"smartlife.iot.common.softaponboarding":{"get_stainfo":{"ssid":"<ssid>","key_type":3,"rssi":-50,"err_code":0}}}`
### Request 2
`{"smartlife.iot.common.timesetting":{"get_timezone":{}},"context":{"source":"<id>"}}`
### Response 2
`{"smartlife.iot.common.timesetting":{"get_timezone":{"index":18,"err_code":0}}}`
### Request 3
`{"smartlife.iot.common.timesetting":{"get_time":{}},"context":{"source":"<id>"}}`
### Response 3
`{"smartlife.iot.common.timesetting":{"get_time":{"year":2020,"month":10,"mday":9,"hour":12,"min":0,"sec":56,"err_code":0}}}`

### Request 
`{"smartlife.iot.lighting_effect":{"set_lighting_effect":{"custom":0,"direction":4,"duration":0,"enable":1,"expansion_strategy":1,"id":"xqUxDhbAhNLqulcuRMyPBmVGyTOyEMEu","name":"Aurora","repeat_times":0,"segments":[0],"sequence":[[120,100,100],[240,100,100],[260,100,100],[280,100,100]],"spread":7,"transition":1500,"type":"sequence"}},"context":{"source":"<id>"}}`
### Response 
`{"smartlife.iot.lighting_effect":{"set_lighting_effect":{"err_code":0}}}`

### PR DESCRIPTION
**Pull request by Hilbrand**
Closes #8709

Additional:
Added action to send and receive json commands to a tplink device.
This can be used for test purposes or to run commands not available through channels.

Changed retry getting values 5 times before setting the device offline..Reduced socket time out to 2 seconds as it normally should react quickly and if it times out it tries again.

### CODE REVIEW

File: bundles/org.openhab.binding.tplinksmarthome/README.md [hunk 1]
```diff
@@ -242,6 +242,24 @@ Switching, Brightness and Color is done using the `color` channel.
...
 
+### KL400 Kasa Smart LED Light Strip
+
+* Power On/Off
+* Fine-tune colors
+* Adjust light appearance from soft white (2500k) to daylight (9000k)
+* Adjust the brightness
+* Wi-Fi signal strength (RSSI)
+
+### KL400 Kasa Smart LED Light Strip, 16 Color Zones
```
Comment (dustinsoftware) [inline bundles/org.openhab.binding.tplinksmarthome/README.md:253]:
```suggestion
### KL430 Kasa Smart LED Light Strip, 16 Color Zones
```

**Review by dustinsoftware — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
@Hilbrand : is it still work in progress ?

**Comment by Hilbrand** [comment]
I've remove the label work-in-progress as it's ready for review/merge.

File: bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/handler/TPLinkSmartHomeActions.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@ThingActionsScope(name = "tplinksmarthome")
+@NonNullByDefault
+public class TPLinkSmartHomeActions implements ThingActions, ThingHandlerService {
+
+    private @Nullable ThingHandler handler;
+
+    @RuleAction(label = "@text/actions.send.label", description = "@text/actions.send.description")
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/handler/TPLinkSmartHomeActions.java:37]: actions.send.label and actions.send.description have to be added in tplinksmarthome.properties

File: bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/handler/TPLinkSmartHomeActions.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * TP-Link Smart Home Rule Actions.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@ThingActionsScope(name = "tplinksmarthome")
+@NonNullByDefault
+public class TPLinkSmartHomeActions implements ThingActions, ThingHandlerService {
+
+    private @Nullable ThingHandler handler;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/handler/TPLinkSmartHomeActions.java:35]: Just a suggestion: You could use type `SmartHomeHandler` and in `setThingHandler` you then check if parameter "handler" is an instance of `SmartHomeHandler`.

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@Hilbrand : my intention was to merge. What did you change after my review ? Why a force push which I have no way to see what you changed since my last review ?

**Comment by Hilbrand** [comment]
@lolodomo I've only squashed to make sure it gets the commit message I preferred. You can see here there are o differences: https://github.com/hilbrand/openhab-addons/compare/fd7c251..5859a02

**Review by lolodomo — APPROVED** [review]

**Comment by dustinsoftware** [comment]
Just FYI by squashing and force pushing, my name on the contribution was lost, would have been cool to keep that on the contributor page. :(

**Comment by lolodomo** [comment]
Sorry for that but unfortunately the squashing was not done by myself.

**Comment by Hilbrand** [comment]
@dustinsoftware that would have been inevitable. Even if I wouldn't have squashed it, it would have happened when @lolodomo would merge this pr. Because all pr's in this repository are squashed before merging.

**Comment by lolodomo** [comment]
@Hilbrand : when squashing, we can keep the mention of several contributors, signed-off- by for the main author and another (sorry I do not remember the exact tag) for additional contributors.

**Comment by Hilbrand** [comment]
@lolodomo Yes adding the other contributors is required, with `Also-by`, as is described on the [openHAB contribution page](https://www.openhab.org/docs/developer/contributing.html#sign-your-work), and that is what I put in the squash commit. But what @dustinsoftware is referring to is the [contributors page](https://github.com/openhab/openhab-addons/graphs/contributors) here on GitHub. If there are multiple contributors when the pr is merged and squashed it will register the commit to the original creator of the pr, and that will show up in git blame and therefor the contributors page.

**Comment by lolodomo** [comment]
Ok, that is clearer now.

**Pull request merged**